src/Form/IiifSettingsForm.php

```diff
@@ -96,32 +93,22 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
     // Don't validate External one, since PHP could probably fail.
-    try {
-      /* @var Client $this->httpClient  */
-      $response = $this->httpClient
-        ->head(
-          $form_state
-            ->getValue('int_server_url')
-        );
-      }
-    catch(ConnectException $exception) {
-      $responseMessage = $exception->getMessage();
-      $form_state->setErrorByName('int_server_url', t("We could not contact your Public IIIF server: @error", [
-        '@error' => $responseMessage
+    $validator = new IiifUrlValidator();
+
+    $internalUrlError = $validator->checkInternalUrl($form_state->getValue('int_server_url'));
+    // todo book p. 303 i think there is an isEmpty() method
```

Comment:
hm I think I was confused because this isEmpty() function is something in the FieldType class. Not relevant here but different from php empty(). Still learning the different levels of content!!